internal/ls/organizeimports.go

```diff
@@ -126,3 +127,287 @@ func isIndexFileName(fileName string) bool {
+func getImportSpecifierInsertionIndex(sortedImports []*ast.Node, newImport *ast.Node, comparer func(s1, s2 *ast.Node) stringutil.Comparison) int {
+	if len(sortedImports) == 0 {
+		return 0
+	}
+
+	low := 0
+	high := len(sortedImports) - 1
+
+	for low <= high {
+		middle := low + ((high - low) >> 1)
+		midKey := sortedImports[middle]
+		switch comparer(midKey, newImport) {
+		case stringutil.ComparisonLessThan:
+			low = middle + 1
+		case stringutil.ComparisonEqual:
+			return middle
+		case stringutil.ComparisonGreaterThan:
+			high = middle - 1
+		}
+	}
+
+	return low
+}
+
+// getOrganizeImportsStringComparerWithDetection detects the string comparer to use based on existing imports
+func getOrganizeImportsStringComparerWithDetection(originalImportDecls []*ast.Statement, preferences *UserPreferences) (comparer func(a, b string) stringutil.Comparison, isSorted bool) {
+	// Get the list of comparers to test based on preferences
+	var comparersToTest []func(a, b string) stringutil.Comparison
+	if preferences != nil && !preferences.OrganizeImportsIgnoreCase.IsUnknown() {
+		// If case sensitivity is explicitly specified, only test that one
+		comparersToTest = []func(a, b string) stringutil.Comparison{
+			getOrganizeImportsOrdinalStringComparer(preferences.OrganizeImportsIgnoreCase.IsTrue()),
+		}
+	} else {
+		// Otherwise, test both case-insensitive and case-sensitive
+		comparersToTest = []func(a, b string) stringutil.Comparison{
+			getOrganizeImportsOrdinalStringComparer(true),  // case-insensitive first (higher priority)
+			getOrganizeImportsOrdinalStringComparer(false), // case-sensitive second
+		}
+	}
+
+	result := detectModuleSpecifierCaseBySort([][]*ast.Statement{originalImportDecls}, comparersToTest)
+	return result.comparer, result.isSorted
+}
+
+type caseSensitivityDetectionResult struct {
+	comparer func(a, b string) stringutil.Comparison
+	isSorted bool
+}
+
+func detectModuleSpecifierCaseBySort(importDeclsByGroup [][]*ast.Statement, comparersToTest []func(a, b string) stringutil.Comparison) caseSensitivityDetectionResult {
+	moduleSpecifiersByGroup := make([][]string, 0, len(importDeclsByGroup))
+	for _, importGroup := range importDeclsByGroup {
+		moduleNames := make([]string, 0, len(importGroup))
+		for _, decl := range importGroup {
+			if expr := getModuleSpecifierExpression(decl); expr != nil {
+				moduleNames = append(moduleNames, getExternalModuleName(expr))
+			} else {
+				moduleNames = append(moduleNames, "")
+			}
+		}
+		moduleSpecifiersByGroup = append(moduleSpecifiersByGroup, moduleNames)
+	}
+	return detectCaseSensitivityBySort(moduleSpecifiersByGroup, comparersToTest)
+}
+
+func detectCaseSensitivityBySort(originalGroups [][]string, comparersToTest []func(a, b string) stringutil.Comparison) caseSensitivityDetectionResult {
+	var bestComparer func(a, b string) stringutil.Comparison
+	bestDiff := int(^uint(0) >> 1) // max int
```

Comment:
Can probably just do `math.MaxInt`.

Reply:
Hah, the organizeImports helpers were a pure agent mode contribution and I didn't catch this